app/main.py

```diff
@@ -1,8 +1,18 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[self.name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    new_people = [(Person(person.get("name"), person.get("age")))
+                  for person in people]
+    for i, person in enumerate(people):
+        if "wife" in person and person.get("wife") is not None:
```

Comment:
```suggestion
        if person.get("wife") is not None:
```
This condition is redundant in this case